sway-lib-core/src/ops.sw

```diff
@@ -121,7 +121,7 @@ pub trait Mod {
 impl Mod for u64 {
     fn modulo(self, other: Self) -> Self {
         asm(r1: self, r2: other, r3) {
-            mod r3 r1 r2;
+            r#mod r3 r1 r2;
```

Comment:
Ah this is pretty annoying. I don't think `asm` blocks should care about keywords as no keywords are actually allowed in there anyways. `asm` blocks should have their own grammar that is completely independent of the rest of Sway.

Reply:
Agreed, I think it's out of scope for this but we definitely should make the `asm` block grammar independent.